packages/react/src/components/Dialog/index.tsx

```diff
@@ -187,6 +195,28 @@ export default class Dialog extends React.Component<DialogProps, DialogState> {
     }
     this.state.isolator?.activate();
   }
+
+  bindKeyboardEvents(event: DialogEvent) {
+    const handleEscape = (keyboardEvent: KeyboardEvent) => {
+      if (!keyboardEvent.defaultPrevented && isEscape(keyboardEvent)) {
```

Comment:
I think one thing we're missing either here or when we bind events is checking for `forceAction`. "Esc" appears to still calling `.close()` and returning focus to the original trigger without actually closing the modal when `forceAction` is set to true.

You can see this behavior in the deploy preview on the `Alert` component.

Reply:
Shoot, I forgot about `forceAction`. I've made some changes to fix that, and added an additional test case to ensure it functions as expected in both states.